Examples/Io/Root/include/ActsExamples/Io/Root/RootTrackStatesWriter.hpp

```diff
@@ -144,6 +144,11 @@ class RootTrackStatesWriter final : public WriterT<ConstTrackContainer> {
   /// truth parameter eT
   std::vector<float> m_t_eT;
 
+  /// event-unique particle identifier a.k.a barcode for hits per each surface
+  std::vector<std::vector<double>> m_particleId; // using double (8 bytes) to save uint64_t (also 8 bytes)
```

Comment:
did you check if that works for your use case? for high long values double will not capture all the information and you won't be able to match the particle

Reply:
hi Andreas, yes, I checked that the particle ids extracted the tree are exactly the same. Both uint64_t and double are 8 bytes, so all information is kept.

>>> Saving step ("values" directly from the Barcodes):
##### new track: nMeas = 11, nStates = 21
volume-layer: 21-6, particle id: 1|0|7|0|0, value = 4503599744811008
volume-layer: 21-4, particle id: 1|0|7|0|0, value = 4503599744811008
volume-layer: 21-2, particle id: 1|0|7|0|0, value = 4503599744811008
volume-layer: 18-10, particle id: 1|0|7|0|0, value = 4503599744811008
volume-layer: 18-8, particle id: 1|0|19|0|0, value = 4503599946137600
volume-layer: 18-6, particle id: 1|0|7|0|0, value = 4503599744811008
volume-layer: 18-4, particle id: 1|0|7|0|0, value = 4503599744811008
volume-layer: 18-2, particle id: 1|0|11|0|0, value = 4503599811919872
volume-layer: 13-6, particle id: 1|0|16|0|0, value = 4503599895805952
volume-layer: 13-4, particle id: 1|0|16|0|0, value = 4503599895805952
volume-layer: 13-2, particle id: 1|0|1|0|0, value = 4503599644147712

>>> Analysis of the tree:
volume-layer-sensitive: 21-6-82
    >> hit 0: particleId = 4503599744811008
volume-layer-sensitive: 21-4-81
    >> hit 0: particleId = 4503599744811008
volume-layer-sensitive: 21-2-81
    >> hit 0: particleId = 4503599744811008
volume-layer-sensitive: 18-10-81
    >> hit 0: particleId = 4503599744811008
volume-layer-sensitive: 18-8-113
    >> hit 0: particleId = 4503599946137600
volume-layer-sensitive: 18-6-113
    >> hit 0: particleId = 4503599744811008
volume-layer-sensitive: 18-4-113
    >> hit 0: particleId = 4503599744811008
volume-layer-sensitive: 18-2-113
    >> hit 0: particleId = 4503599811919872
volume-layer-sensitive: 13-6-1
    >> hit 0: particleId = 4503599895805952
volume-layer-sensitive: 13-4-1
    >> hit 0: particleId = 4503599895805952
volume-layer-sensitive: 13-2-1
    >> hit 0: particleId = 4503599644147712

Reply:
it is true that both are 8 byte but numbers are represented differently with long and double. digits in doubles are represented in the mantissa and and exponent will shift then comma. by converting long to double you will only utilize part of the exponent and all the mantissa which is not enough space to carry all the information.

that said as long as the number is small enough and the rounding plays nice it "should" work

Reply:
you are right.. 
I checked that double can manage the barcodes with 15000 particles in the detector, the barcode.value distribution looks like this
<img width="724" alt="image" src="https://github.com/acts-project/acts/assets/12641156/155ca744-bfa3-4b08-a20c-4e681126248e">

Reply:
I would prefer the `double` solution for now. It is not optimal but we do similar stuff to surround root related problems in the Examples already

Reply:
Would an alternative be to store the id in two 32bit ints?

Reply:
hi Paul,
unfortunately, no..
Error in <TTree::Branch>: The class requested (vector<vector<unsigned int> >) for the branch "particle_idsTest" is an instance of an stl collection and does not have a compiled CollectionProxy.
...
Error in <TTree::Branch>: The class requested (vector<vector<int> >) for the branch "particle_idsTest2" is an instance of an stl collection and does not have a compiled CollectionProxy.